### PR TITLE
Change the method 'is_node_rack_or_zone_exist' to get the node name instead of the node object 4.11 #8344

### PR DESCRIPTION
**Pull request by yitzhak12**
Backport PR https://github.com/red-hat-storage/ocs-ci/pull/8344.

### CODE REVIEW

**Review by am-agrawa — DISMISSED** [review]

**Review by amr1ta — DISMISSED** [review]
Please run the pr validation job to get it merged.

**Comment by am-agrawa** [comment]
@yitzhak12 Could you pls merge it asap? The test is failing with latest 4.11 runs too- https://url.corp.redhat.com/d4234a7

File: ocs_ci/ocs/node.py [hunk 1]
```diff
@@ -2667,3 +2667,119 @@ def generate_nodes_for_provider_worker_node_tests():
...
+    factory = platform_nodes.PlatformNodesFactory()
+    nodes = factory.get_nodes_platform()
+    waiting_time = 30
+    for node in node_objs:
+        node_name = node.name
+        unschedule_nodes([node_name])
+        drain_nodes([node_name])
+        nodes.restart_nodes([node], wait=False)
+        log.info(f"Waiting for {waiting_time} seconds")
+        time.sleep(waiting_time)
```
Comment (ebenahar) [inline ocs_ci/ocs/node.py:2690]: can't we use a function to wait for node connectivity instead?

Comment (yitzhak12) [reply ocs_ci/ocs/node.py:2690]: Yes, it makes sense. But this function was initially created by Akarsha Rai: https://github.com/red-hat-storage/ocs-ci/blame/master/ocs_ci/ocs/node.py#L2722. I needed to rebase from the `release-4.11`. That's why the change is included in the current PR. I can make another PR from the `master` branch and change the above code.

**Review by ebenahar — APPROVED** [review]

**Review by prsurve — APPROVED** [review]

**Pull request merged**